actionpack/lib/action_dispatch/http/response.rb

```diff
@@ -73,11 +64,11 @@ def to_hash
     #   headers["Content-Type"] = "application/json"
     #   headers["Content-Type"] # => "application/json"
     #
-    attr_reader :header
+    attr_reader :headers
 
-    alias_method :headers,  :header
+    alias_method :header, :headers
 
-    delegate :[], :[]=, to: :@header
+    delegate :[], :[]=, to: :@headers
```

Comment:
These `header`/`headers` local and instance variable and and `Header`/`Headers` constant name changes don't seem to be necessary for the PR.  Not opposed to them eventually, but this PR is large enough without them. It's best to not mix unnecessary stylistic changes with structural changes to add new features, IMO, as it makes review significantly more challenging.

Reply:
I agree, so I removed it and will re-roll this later.